### ISSUE TITLE
Thread Sanitizer Data race in [GDTCCTUploadOperation finishOperation]

### ISSUE THREAD

**Issue by KoCMoHaBTa** [google/GoogleDataTransport#14]
I've just ran my code trough Thread Sanitizer and got report for data race in `GDTCCTUploadOperation`

We use the following Pods
```
  pod 'Firebase/Core'
  pod 'Firebase/Performance'
  pod 'Firebase/Messaging'
  pod 'Firebase/Analytics'
  pod 'Firebase/Crashlytics'
```

Already updated to latest versions:
```
Installing Firebase 7.9.0 (was 7.8.1)
Installing FirebaseABTesting 7.9.0 (was 7.8.0)
Installing FirebaseAnalytics 7.9.0 (was 7.8.1)
Installing FirebaseCore 7.9.0 (was 7.8.0)
Installing FirebaseCoreDiagnostics 7.9.0 (was 7.8.0)
Installing FirebaseCrashlytics 7.9.0 (was 7.8.0)
Installing FirebaseInstallations 7.9.0 (was 7.8.0)
Installing FirebaseInstanceID 7.9.0 (was 7.8.0)
Installing FirebaseMessaging 7.9.0 (was 7.8.0)
Installing FirebasePerformance 7.9.0 (was 7.8.0)
Installing FirebaseRemoteConfig 7.9.0 (was 7.8.0)
Installing GoogleAppMeasurement 7.9.0 (was 7.8.1)
```

In code we have custom Crashlytics logs and custom analytics event and screen tracking.
The project uses Swift + SwiftUI and targets iOS 14.
The issue has been detected on iOS 14 Simulator.

Console output:
```
WARNING: ThreadSanitizer: data race (pid=72528)
  Write of size 1 at 0x7b44000901f9 by thread T18:
    #0 -[GDTCCTUploadOperation finishOperation] GDTCCTUploadOperation.m:508 (GoogleDataTransport:x86_64+0x11821)
    #1 __53-[GDTCCTUploadOperation uploadTarget:withConditions:]_block_invoke.72 GDTCCTUploadOperation.m:184 (GoogleDataTransport:x86_64+0xccdf)
    #2 __44-[FBLPromise(CatchAdditions) onQueue:catch:]_block_invoke FBLPromise+Catch.m:34 (FBLPromises:x86_64+0xbf1e)
    #3 __56-[FBLPromise chainOnQueue:chainedFulfill:chainedReject:]_block_invoke.68 FBLPromise.m:277 (FBLPromises:x86_64+0x2378f)
    #4 __44-[FBLPromise observeOnQueue:fulfill:reject:]_block_invoke_2 FBLPromise.m:229 (FBLPromises:x86_64+0x22738)
    #5 __wrap_dispatch_group_async_block_invoke <null>:2 (libclang_rt.tsan_iossim_dynamic.dylib:x86_64+0x71c37)
    #6 _dispatch_client_callout <null>:2 (libdispatch.dylib:x86_64+0x39c7)

  Previous read of size 1 at 0x7b44000901f9 by thread T16:
    #0 -[GDTCCTUploadOperation isFinished] GDTCCTUploadOperation.m:489 (GoogleDataTransport:x86_64+0x11cec)
    #1 __NSOQSchedule_f <null>:2 (Foundation:x86_64+0xddcee)
    #2 _dispatch_client_callout <null>:2 (libdispatch.dylib:x86_64+0x39c7)

  Location is heap block of size 320 at 0x7b4400090100 allocated by thread T16:
    #0 calloc <null>:2 (libclang_rt.tsan_iossim_dynamic.dylib:x86_64+0x4f072)
    #1 _objc_rootAllocWithZone <null>:2 (libobjc.A.dylib:x86_64+0x16a99)
    #2 __52-[GDTCORUploadCoordinator uploadTargets:conditions:]_block_invoke GDTCORUploadCoordinator.m:112 (GoogleDataTransport:x86_64+0x4352b)
    #3 __tsan::invoke_and_release_block(void*) <null>:2 (libclang_rt.tsan_iossim_dynamic.dylib:x86_64+0x70ebb)
    #4 _dispatch_client_callout <null>:2 (libdispatch.dylib:x86_64+0x39c7)

  Thread T18 (tid=3005412, running) is a GCD worker thread

  Thread T16 (tid=3004697, running) is a GCD worker thread

SUMMARY: ThreadSanitizer: data race GDTCCTUploadOperation.m:508 in -[GDTCCTUploadOperation finishOperation]
```

Screenshots:
![Screen Shot 2021-03-27 at 9 54 05](https://user-images.githubusercontent.com/1181346/112714472-25554e00-8ee3-11eb-901d-ea855882bb96.png)
![Screen Shot 2021-03-27 at 9 54 20](https://user-images.githubusercontent.com/1181346/112714473-26867b00-8ee3-11eb-81b7-93ca617adc15.png)
![Screen Shot 2021-03-27 at 9 54 28](https://user-images.githubusercontent.com/1181346/112714474-271f1180-8ee3-11eb-82d4-0250e5dfcd96.png)
![Screen Shot 2021-03-27 at 9 54 37](https://user-images.githubusercontent.com/1181346/112714471-238b8a80-8ee3-11eb-86b1-ba68aac32a89.png)

**Comment by paulb777** [comment]
Thanks for the report. These should be fixed by #11 and we'll see if we can publish an update in the next few days.

**Comment by maksymmalyhin** [comment]
The should be fixed in GoogleDataTransport 8.3.1. Please try to update and let us know if you still have the issue.

**Comment by KoCMoHaBTa** [comment]
Thanks for the update, it appears to be fixed.
